src/engine.cc

```diff
@@ -0,0 +1,94 @@
+#include "engine.h"
+#include "gobject.h"
+
+#include <eos-knowledge-content.h>
+
+using namespace v8;
+
+struct CallbackData {
+  Persistent<Function> resolve;
+  Persistent<Function> reject;
+};
+
+static void EngineGetObjectCallback(GObject *source,
+                                    GAsyncResult *result,
+                                    gpointer user_data)
+{
+  Isolate *isolate = Isolate::GetCurrent ();
+  HandleScope scope(isolate);
+  CallbackData *data = (CallbackData *)user_data;
```

Comment:
In GJS I've started to use stuff like `auto data = static_cast<CallbackData *>(user_data)` so that (1) the type name is not specified twice and (2) potentially dangerous casts are made obvious by using C++ style.